test/e2e/network/service.go

```diff
@@ -101,42 +101,87 @@ var (
 // portsByPodName is a map that maps pod name to container ports.
 type portsByPodName map[string][]int
 
+// affinityCheckerPollingArgs is a helper function type used by checkAffinity.
+// The returned function returns polling interval and timeout and function returning
+// a set of hosts that were pinged by the checker.
+type affinityCheckerPollingArgs func() (time.Duration, time.Duration, func() []string)
+
+func affinityCheckFromPod(execPod *v1.Pod, serviceIP string, servicePort int) affinityCheckerPollingArgs {
+	timeout := AffinityTimeout
+	// interval considering a maximum of 2 seconds per connection
+	interval := 2 * AffinityConfirmCount * time.Second
+
+	serviceIPPort := net.JoinHostPort(serviceIP, strconv.Itoa(servicePort))
+	curl := fmt.Sprintf(`curl -q -s --connect-timeout 2 http://%s/`, serviceIPPort)
+	cmd := fmt.Sprintf("for i in $(seq 0 %d); do echo; %s ; done", AffinityConfirmCount, curl)
+	getHosts := func() []string {
+		stdout, err := framework.RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		if err != nil {
+			framework.Logf("Failed to get response from %s. Retry until timeout", serviceIPPort)
+			return nil
+		}
+		return strings.Split(stdout, "\n")
+	}
+
+	return func() (time.Duration, time.Duration, func() []string) {
+		return interval, timeout, getHosts
+	}
```

Comment:
weird to have these return a function that returns those arguments, rather than just returning the arguments themselves

Reply:
the function itself is slightly longer then, but I'm fine with changing too.
Done